android/src/main/java/com/novoda/notils/devicedetection/X86Detector.java

```diff
@@ -1,18 +1,40 @@
 package com.novoda.notils.devicedetection;
 
+import android.annotation.TargetApi;
+import android.os.Build;
+
 import java.util.Arrays;
 
-class X86Detector {
+class X86Detector implements Detector {
 
     private static final String X86 = "x86";
 
     private final String[] supportedAbis;
 
+    public static Detector newInstance() {
```

Comment:
:+1: I think this is clearer, keeping this internal.

Not sure the interface adds anything though.